src/main/scala/cromwell/binding/values/WdlInteger.scala

```diff
@@ -1,29 +1,73 @@
+      case _ => invalid(s"$value + $rhs")
     }
   }
-  def subtract(rhs: WdlValue): WdlValue = ???
-  def multiply(rhs: WdlValue): WdlValue = ???
-  def divide(rhs: WdlValue): WdlValue = ???
-  def mod(rhs: WdlValue): WdlValue = ???
-  def equals(rhs: WdlValue): WdlValue = ???
-  def notEquals(rhs: WdlValue): WdlValue = ???
-  def lessThan(rhs: WdlValue): WdlValue = ???
-  def lessThanOrEqual(rhs: WdlValue): WdlValue = ???
-  def greaterThan(rhs: WdlValue): WdlValue = ???
-  def greaterThanOrEqual(rhs: WdlValue): WdlValue = ???
-  def or(rhs: WdlValue): WdlValue = ???
-  def and(rhs: WdlValue): WdlValue = ???
-  def not: WdlValue = ???
-  def unaryPlus: WdlValue = ???
-  def unaryMinus: WdlValue = ???
-  def asString = value.toString
+  override def subtract(rhs: WdlValue): WdlValue = {
+    rhs match {
+      case r:WdlInteger => WdlInteger(value - r.value)
+      case r:WdlFloat => WdlFloat(value - r.value)
+      case _ => invalid(s"$value - $rhs")
+    }
+  }
+  override def multiply(rhs: WdlValue): WdlValue = {
+    rhs match {
+      case r:WdlInteger => WdlInteger(value * r.value)
+      case r:WdlFloat => WdlFloat(value * r.value)
+      case _ => invalid(s"$value * $rhs")
+    }
+  }
+  override def divide(rhs: WdlValue): WdlValue = {
+    rhs match {
+      case r:WdlInteger if r.value == 0 => throw new WdlExpressionException(s"Divide by zero error: $value / $rhs")
+      case r:WdlInteger => WdlInteger(value / r.value)
+      case r:WdlFloat if r.value == 0.toFloat => throw new WdlExpressionException(s"Divide by zero error: $value / $rhs")
+      case r:WdlFloat => WdlFloat(value / r.value)
+      case _ => invalid(s"$value / $rhs")
+    }
+  }
+  override def mod(rhs: WdlValue): WdlValue = {
+    rhs match {
+      case r:WdlInteger if r.value == 0 => throw new WdlExpressionException(s"Divide by zero error: $value / $rhs")
+      case r:WdlInteger => WdlInteger(value % r.value)
+      case r:WdlFloat if r.value == 0.toFloat => throw new WdlExpressionException(s"Divide by zero error: $value / $rhs")
+      case r:WdlFloat => WdlFloat(value % r.value)
+      case _ => invalid(s"$value % $rhs")
+    }
+  }
+  override def equals(rhs: WdlValue): WdlBoolean = {
+    rhs match {
+      case r:WdlInteger => WdlBoolean(value == r.value)
+      case r:WdlFloat => WdlBoolean(value == r.value)
+      case _ => invalid(s"$value == $rhs")
+    }
+  }
+  override def lessThan(rhs: WdlValue): WdlBoolean = {
+    rhs match {
+      case r:WdlInteger => WdlBoolean(value < r.value)
+      case r:WdlFloat => WdlBoolean(value < r.value)
+      case _ => invalid(s"$value < $rhs")
+    }
+  }
+  override def greaterThan(rhs: WdlValue): WdlBoolean = {
+    rhs match {
+      case r:WdlInteger => WdlBoolean(value > r.value)
+      case r:WdlFloat => WdlBoolean(value > r.value)
+      case _ => invalid(s"$value > $rhs")
+    }
+  }
+  override def greaterThanOrEqual(rhs: WdlValue): WdlBoolean = WdlBoolean(equals(rhs).value || greaterThan(rhs).value)
```

Comment:
Woops this probably shouldn't be there because it should be defined on the `WdlTrait`